Src/Web/Web.Shared.Net/RequestTrackingTelemetryModule.cs

```diff
@@ -347,5 +382,151 @@ private bool TryInitializeCorrelationHelperIfNotInitialized()
+#if DEBUG
+            private const string HeaderParentRequestId = "ApplicationInsights-RequestTrackingTelemetryModule-ParentRequest-Id";
+            private const string HeaderRequestId = "ApplicationInsights-RequestTrackingTelemetryModule-Request-Id";
+#endif
+
+            private static object semaphore = new object();
+
+            /// <summary>
+            /// Using this as a hash-set of current active requests. The second value is not used.
+            /// </summary>
+            private static ConcurrentDictionary<string, bool> activeRequestsA = new ConcurrentDictionary<string, bool>(System.Environment.ProcessorCount, DEFAULTMAXVALUE);
+            private static ConcurrentDictionary<string, bool> activeRequestsB = new ConcurrentDictionary<string, bool>(System.Environment.ProcessorCount, DEFAULTMAXVALUE);
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="ChildRequestTrackingSuppressionModule" /> class.
+            /// </summary>
+            /// <param name="maxRequestsTracked">The maximum number of active requests to be tracked before resetting the dictionary.</param>
+            internal ChildRequestTrackingSuppressionModule(int maxRequestsTracked = DEFAULTMAXVALUE)
+            {
+                this.MAXSIZE = maxRequestsTracked > 0 ? maxRequestsTracked : DEFAULTMAXVALUE;
+            }
+
+            /// <summary>
+            /// Gets the Max number of request ids to cache.
+            /// </summary>
+            internal int MAXSIZE { get; private set; }
+
+            /// <summary>
+            /// Request will be tagged with an id to identify if it should be logged later.
+            /// </summary>
+            internal void OnBeginRequest_IdRequest(HttpContext context)
+            {
+                this.TagRequest(context);
+            }
+
+            /// <summary>
+            /// OnEndRequest - Should this request be logged?
+            /// Will compare a request id against a hash-set of known requests.
+            /// If this request is not known, add it to hash-set and return true (safe to log).
+            /// If this request is known, return false (do not log twice).
+            /// Additional requests with the same id will return false.
+            /// </summary>
+            internal bool OnEndRequest_ShouldLog(HttpContext context)
+            {
+                var headers = context.Request.Headers;
+
+                var rootRequestId = headers[HeaderRootRequestId];
+                if (rootRequestId != null)
+                {
+                    if (!this.IsRequestKnown(rootRequestId))
+                    {
+                        // doesn't exist add to dictionary and return true
+                        this.AddRequestToDictionary(rootRequestId);
+                        return true;
+                    }
+                }
+
+                return false;
```

Comment:
I strongly recommend writing something to eventlog whenever we are dropping requests. Without this, it'll be quite difficult if some issues occurs about 'requests missing'.

Reply:
The above return false occurs if request is not tagged. So please add log about it.